src/vs/workbench/browser/parts/editor/editorActions.ts

```diff
@@ -1913,6 +1915,28 @@ export class ReopenResourcesAction extends Action {
 
 		const options = activeEditorPane.options;
 		const group = activeEditorPane.group;
+		// The current editor needs to be saved before it can be triggered with open with
+		if (activeInput.isDirty()) {
```

Comment:
Will this fire even if you select the default VS Code editor? Also, should it only be fired if you select a custom binary editor since a custom text editor should be able to handle this case correctly?

Reply:
Mainly wondering if it should be moved into the call to `openEditorWith` instead after you have selected an editor but before it has actually replaced the old editor